src/actions/posts.ts

```diff
@@ -206,8 +206,8 @@ export function createPost(post: Post, files: any[] = []) {
         dispatch({
             type: PostTypes.RECEIVED_NEW_POST,
             data: {
-                id: pendingPostId,
```

Comment:
These had to change because TypeScript expected `newPost` to be a complete Post object which meant that it thought this ID was being overwritten.

The only case where this ID was being overwritten was in the corresponding test that I changed, and that test was invalid since the request was never actually made to create the post. To fix that, I changed it to use `createPostImmediately` instead which does send the request, and then I had the server respond with the post ID instead.